docs/experiments/001_dbt_starrock_external_iceberg_catalog_sources.md

```diff
@@ -0,0 +1,157 @@
+
+1. Identify the source dataset.
+2. Load the source data into an Iceberg table. 
+3. Run `dbt build` (combination of `run` and `test`) to transform the data and test it.
+
+![001_flow.png](resources/001_flow.png)
+
+### Tools and Resources
+
+#### Environment & Tools
+
+The following diagram presents the local experimental setup.
+> A `docker-compose.yml` file was used to specify required services. 
+
+![001_stack.png](resources/001_stack.png)
+
+The experimental stack is composed of the following tech/tools: 
+
+- `MinIO` (https://min.io/): Distributed object store.
+- `Iceberg` (https://iceberg.apache.org/): Open table format.
+- `StarRocks` (https://www.starrocks.io/): Analytical data warehouse.
+- `dbt` (https://www.getdbt.com/): Data transformation. 
+- `python` (https://www.python.org/): Used to build scripts to load/clean the data.
+
+#### Data
+
+Re-use the `dbt` tutorial's [Jaffle Shop data](https://github.com/dbt-labs/jaffle-shop?tab=readme-ov-file#-load-the-data-from-s3) as the raw data to be loaded into Iceberg tables.
+
+We use the following files:
+
+  - `raw_customers.csv`
+  - `raw_orders.csv`
+  - `raw_payments.csv`
+
+## Observations
+
+Running `dbt run` produced the following data hierarchy, observable through `dbt docs`:
+
+![001_db_structure.png](resources/001_db_structure.png)
+
+Where:
+- `iceberg.jaffle_shop` comes from the external catalog 
+- `exp001_iceberg_external_sources` is defined as the StarRocks schema
+
+And the following `dbt` relationship graph:
+
+![001_dbt_graph.png](resources/001_dbt_graph.png)
+
+And the materialization was tested for both `materialized: table` : 
+
+![001_table_materialization.png](resources/001_table_materialization.png)
+
+and `materialized: view`: 
+
+![001_view_materialization.png](resources/001_view_materialization.png)
+
+**Logs**:
+
+`dbt run` produced the following logs. 
+
+```
+19:28:19  Running with dbt=1.8.9
+19:28:19  Registered adapter: starrocks=1.7.0
+19:28:20  Found 5 models, 20 data tests, 3 sources, 419 macros
+19:28:20  
+19:28:20  Concurrency: 1 threads (target='dev')
+19:28:20  
+19:28:20  1 of 5 START sql view model exp001_iceberg_external_sources.stg_customers ...... [RUN]
+19:28:20  1 of 5 OK created sql view model exp001_iceberg_external_sources.stg_customers . [SUCCESS 0 in 0.42s]
+19:28:20  2 of 5 START sql view model exp001_iceberg_external_sources.stg_orders ......... [RUN]
+19:28:20  2 of 5 OK created sql view model exp001_iceberg_external_sources.stg_orders .... [SUCCESS 0 in 0.13s]
+19:28:20  3 of 5 START sql view model exp001_iceberg_external_sources.stg_payments ....... [RUN]
+19:28:20  3 of 5 OK created sql view model exp001_iceberg_external_sources.stg_payments .. [SUCCESS 0 in 0.13s]
+19:28:20  4 of 5 START sql table model exp001_iceberg_external_sources.customers ......... [RUN]
+19:31:25  4 of 5 OK created sql table model exp001_iceberg_external_sources.customers .... [SUCCESS 100 in 184.24s] 
```

Comment:
When changing to `materialized: table`, we hit this ~180s delay on the first call. 

There's something going but I'm not sure what :thinking: 
I will investigate this in another experiment as a follow-up question.